BlobStorage/Advanced.cs

```diff
@@ -1277,95 +1054,55 @@ private static async Task ListBlobsHierarchicalListingAsync(CloudBlobContainer c
-        /// </summary>
-        /// <param name="container">The container.</param>
-        /// <param name="blobName">The blob name.</param>
-        /// <returns>A Task object.</returns>
-        private static void GetExistingBlobReference(CloudBlobContainer container, string blobName)
-        {
-            try
-            {
-                // Get a reference to a blob with a request to the server.
-                // If the blob does not exist, this call will fail with a 404 (Not Found).
-                ICloudBlob blob = container.GetBlobReferenceFromServer(blobName);
-
-                // The previous call gets the blob's properties, so it's not necessary to call FetchAttributes
-                // to read a property.
-                Console.WriteLine("Blob {0} was last modified at {1} local time.", blobName,
-                    blob.Properties.LastModified.Value.LocalDateTime);
-            }
-            catch (StorageException e)
-            {
-                if (e.RequestInformation.HttpStatusCode == 404)
-                {
-                    Console.WriteLine("Blob {0} does not exist.", blobName);
-                    Console.WriteLine("Additional error information: " + e.Message);
-                }
-                else
-                {
-                    Console.WriteLine(e.Message);
-                    Console.ReadLine();
-                    throw;
-                }
-            }
-
-            Console.WriteLine();
-        }
 
         /// <summary>
         /// Gets a reference to a blob by making a request to the service.
         /// </summary>
         /// <param name="container">The container.</param>
         /// <param name="blobName">The blob name.</param>
         /// <returns>A Task object.</returns>
-        private static async Task GetExistingBlobReferenceAsync(CloudBlobContainer container, string blobName)
+        private static async Task GetExistingBlobReferenceAsync(BlobContainerClient container, string blobName)
         {
             try
             {
-                // Get a reference to a blob with a request to the server.
-                // If the blob does not exist, this call will fail with a 404 (Not Found).
-                ICloudBlob blob = await container.GetBlobReferenceFromServerAsync(blobName);
+                // Get a reference to a blob.
+                BlobClient blob = container.GetBlobClient(blobName);
 
                 // The previous call gets the blob's properties, so it's not necessary to call FetchAttributes
                 // to read a property.
                 Console.WriteLine("Blob {0} was last modified at {1} local time.", blobName,
-                    blob.Properties.LastModified.Value.LocalDateTime);
+                  (await blob.GetPropertiesAsync()).Value.LastModified.LocalDateTime);
             }
-            catch (StorageException e)
+            catch (RequestFailedException e)
```

Comment:
Use exception filtering here. The `catch` clause can become:

```csharp
catch (RequestFailedException e) when (e.Status == 400)
```

After doing this, remove the `if` statement inside of the `catch` block.